ddtrace/tracer/transport.go

```diff
@@ -52,6 +52,8 @@ type transport interface {
 	// send sends the payload p to the agent using the transport set up.
 	// It returns a non-nil response body when no error occurred.
 	send(p *payload) (body io.ReadCloser, err error)
+	endpoint() string
+	testConn() error
```

Comment:
I think we don't need this one. Using `endpoint()` will be enough to have the `agentReachable` function again, right?